include/etl/fsm.h

```diff
@@ -360,8 +345,25 @@ namespace etl
       p_state = nullptr;
     }
 
+
   private:
 
+    void transition(etl::fsm_state_id_t next_state_id)
```

Comment:
This looks like it should work, although I can see that before there were subtle differences between the transition into the first state and transitions to subsequent states. I like the fact that it's now consolidated, but I didn't _really_ investigate to confirm that the logic is equivalent. I assume you've verified it's behaving correctly.

Reply:
the logic is a little different now - before, the `on_exit_state` was inexplicably not called from the `start` method's transition logic, but it was in `receive`. I also changed the logic so that returning the same state ID **_will_** trigger a transition, whereas before it was the end condition. the new end condition is to return `stay()` instead.

Reply:
to me, it seemed like a weird bug that `start` and `receive` acted almost the same, but not quite